### PR TITLE
fix: change nofile ulimit to string

### PR DESCRIPTION
**Pull request by UtheMan**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/agent/baker.go [hunk 1]
```diff
@@ -448,8 +448,8 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 			if ulimitConfig.MaxLockedMemory != "" {
 				sb.WriteString(fmt.Sprintf("LimitMEMLOCK=%s\n", ulimitConfig.MaxLockedMemory))
 			}
-			if ulimitConfig.NoFile != nil {
-				sb.WriteString(fmt.Sprintf("LimitNOFILE=%d\n", *ulimitConfig.NoFile))
+			if ulimitConfig.NoFile != "" {
```
Comment (cameronmeissner) [inline pkg/agent/baker.go:451]: unrelated, but do we have tests in baker_test.go for this?

**Review by cameronmeissner — APPROVED** [review]

File: pkg/agent/baker.go (hunk 1 above)
Comment (cameronmeissner) [reply pkg/agent/baker.go:451]: mainly for ulimit stuff in general I mean

Comment (UtheMan) [reply pkg/agent/baker.go:451]: Not right now, only e2e. The only custom linux os config tests i could find were for ubuntu 16.04 (probably some old edge case?), so I didn't add any yet for my change. Could add more of them in the future when more options will be added
